core/arch/riscv/plat-spike/drivers/htif.c

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2022 NXP
+ */
+
+#include <compiler.h>
+#include <kernel/spinlock.h>
+#include <stdint.h>
+
+#include "htif.h"
+
+static unsigned int htif_global_lock __nex_bss = SPINLOCK_UNLOCK;
+
+uint64_t tohost __htif;
+uint64_t fromhost __htif;
+
+void htif_lock_global(void)
+{
+	cpu_spin_lock(&htif_global_lock);
+}
+
+void htif_unlock_global(void)
+{
+	cpu_spin_unlock(&htif_global_lock);
+}
+
+static void __maybe_unused tohost_cmd(uintptr_t dev,
```

Comment:
Why are you using the `uintptr_t` type for all the arguments? None of the arguments carry any address.